lib/screens/explore_screens/unplant_seeds/interactor/viewmodels/unplant_seeds_state.dart

```diff
@@ -64,23 +67,26 @@ class UnplantSeedsState extends Equatable {
     TokenDataModel? unplantedInputAmount,
     TextEditingController? controller,
     PageCommand? pageCommand,
+    bool? showUnclaimedBalance,
   }) {
     return UnplantSeedsState(
-        pageState: pageState ?? this.pageState,
-        ratesState: ratesState ?? this.ratesState,
-        onFocus: onFocus ?? this.onFocus,
-        plantedBalance: plantedBalance ?? this.plantedBalance,
-        plantedBalanceFiat: plantedBalanceFiat ?? this.plantedBalanceFiat,
-        isUnplantSeedsButtonEnabled: isUnplantSeedsButtonEnabled ?? this.isUnplantSeedsButtonEnabled,
-        showOverBalanceAlert: showOverBalanceAlert ?? this.showOverBalanceAlert,
-        showMinPlantedBalanceAlert: showMinPlantedBalanceAlert ?? this.showMinPlantedBalanceAlert,
-        unplantedInputAmountFiat: unplantedInputAmountFiat ?? this.unplantedInputAmountFiat,
-        unplantedInputAmount: unplantedInputAmount ?? this.unplantedInputAmount,
-        controller: controller ?? this.controller,
-        pageCommand: pageCommand);
+      pageState: pageState ?? this.pageState,
+      ratesState: ratesState ?? this.ratesState,
+      onFocus: onFocus ?? this.onFocus,
+      plantedBalance: plantedBalance ?? this.plantedBalance,
+      plantedBalanceFiat: plantedBalanceFiat ?? this.plantedBalanceFiat,
+      isUnplantSeedsButtonEnabled: isUnplantSeedsButtonEnabled ?? this.isUnplantSeedsButtonEnabled,
+      showOverBalanceAlert: showOverBalanceAlert ?? this.showOverBalanceAlert,
+      showMinPlantedBalanceAlert: showMinPlantedBalanceAlert ?? this.showMinPlantedBalanceAlert,
+      unplantedInputAmountFiat: unplantedInputAmountFiat ?? this.unplantedInputAmountFiat,
+      unplantedInputAmount: unplantedInputAmount ?? this.unplantedInputAmount,
+      controller: controller ?? this.controller,
```

Comment:
A TextEditingController in the bloc state class?? That is new for me.

Reply:
Yes I was rigth.
From Felix Angelov (author of flutter_bloc) https://github.com/felangel/bloc/issues/2293:

> I would highly recommend against maintaining TextEditingController as part of the bloc. Blocs should ideally be platform-agnostic and have no dependency on Flutter. If you need to use TextEditingControllers I would recommend creating a StatefulWidget and maintaining them as part of the State class. Then you can interface with the control in response to state changes via BlocListener

Reply:
Also how you can do dispose() to the controller in that way?

Reply:
Good find Raul, I will fix this on another pr.

Reply:
this was a miss on me. No flutter code in the blocs, only dart code